### PR TITLE
Change output stream format to H264.RemoteStream

### PR DESCRIPTION
**Pull request by sgfn**
Since the new H264 parser does not accept `RemoteStream` with `type: :packetized`, a suggestion has been made to change it to `H264.RemoteStream` with `alignment: :nalu`. [This will necessitate changes in `membrane_rtp_plugin`.](https://github.com/membraneframework/membrane_rtp_plugin/pull/145)

Discussion of this proposed solution is strongly encouraged below 😄

### CODE REVIEW

**Comment by bblaszkow06** [comment]
It doesn't seem right to change it here - the `H264.RemoteStream` was created to allow passing `AVCDecoderConfigurationRecord` with SPS and PPS, so:
1. The new parser seems to be bugged and ignores the `:decoder_configuration_record` from `H264.RemoteStream`. That record should be parsed as well to properly handle such format
2. Should handle `%RemoteStream{content_format: H264, type: :packetized}` the way it currently handles `%H264.RemoteStream{alignment: :nalu}`

To sum up - IMO the changes should be done in the parser, not here (cc: @varsill @DominikWolek)

**Comment by mickel8** [comment]
On our internal meeting we decided that this is the way to go so I am approving

**Review by mickel8 — APPROVED** [review]

**Review by bblaszkow06 — APPROVED** [review]
